pip install fails on 3.7, prevents Jupyter install

Installation fails with error:
`winpty/cywinpty.c(598): fatal error C1083: Cannot open include file: 'winpty.h': No such file or directory` 

```
(venv3) pip install pywinpty
Collecting pywinpty
  Using cached https://files.pythonhosted.org/packages/29/6c/c207ba85325e7d3fa5df3e2b51954705bb378de54065ee01ba89317c790d/pywinpty-0.5.4.tar.gz
Installing collected packages: pywinpty
  Running setup.py install for pywinpty ... error
    Complete output from command C:\Project\venv3\scripts\python.exe -u -c "import setuptools, tokenize;__file__='C:\\Users\\USER\\AppData\\Local\\Temp\\pip-install-5xcqjygg\\pywinpty\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record C:\Users\USER\AppData\Local\Temp\pip-record-74im8oyr\install-record.txt --single-version-externally-managed --compile --install-headers C:\Project\venv3\include\site\python3.7\pywinpty:
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-3.7
    creating build\lib.win-amd64-3.7\winpty
    copying winpty\ptyprocess.py -> build\lib.win-amd64-3.7\winpty
    copying winpty\winpty_wrapper.py -> build\lib.win-amd64-3.7\winpty
    copying winpty\__init__.py -> build\lib.win-amd64-3.7\winpty
    creating build\lib.win-amd64-3.7\winpty\tests
    copying winpty\tests\test_cywinpty.py -> build\lib.win-amd64-3.7\winpty\tests
    copying winpty\tests\test_ptyprocess.py -> build\lib.win-amd64-3.7\winpty\tests
    copying winpty\tests\test_winpty_wrapper.py -> build\lib.win-amd64-3.7\winpty\tests
    copying winpty\tests\__init__.py -> build\lib.win-amd64-3.7\winpty\tests
    creating build\lib.win-amd64-3.7\winpty\_winpty
    copying winpty\_winpty\__init__.py -> build\lib.win-amd64-3.7\winpty\_winpty
    running build_ext
    building 'winpty.cywinpty' extension
    creating build\temp.win-amd64-3.7
    creating build\temp.win-amd64-3.7\Release
    creating build\temp.win-amd64-3.7\Release\winpty
    C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.15.26726\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -IC:\Project\venv3\include -IC:\Python37\include -IC:\Python37\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.15.26726\ATLMFC\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.15.26726\include" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17134.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17134.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17134.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17134.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17134.0\cppwinrt" /Tcwinpty/cywinpty.c /Fobuild\temp.win-amd64-3.7\Release\winpty/cywinpty.obj
    cywinpty.c
    winpty/cywinpty.c(598): fatal error C1083: Cannot open include file: 'winpty.h': No such file or directory
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\VC\\Tools\\MSVC\\14.15.26726\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2

    ----------------------------------------
Command "C:\Project\venv3\scripts\python.exe -u -c "import setuptools, tokenize;__file__='C:\\Users\\USER\\AppData\\Local\\Temp\\pip-install-5xcqjygg\\pywinpty\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record C:\Users\USER\AppData\Local\Temp\pip-record-74im8oyr\install-record.txt --single-version-externally-managed --compile --install-headers C:\Project\venv3\include\site\python3.7\pywinpty" failed with error code 1 in C:\Users\USER\AppData\Local\Temp\pip-install-5xcqjygg\pywinpty\
```

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -59,8 +59,8 @@ install:
   # Use test conda environment
   - "activate test"
   # - CALL "C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\vcvarsall.bat" %ARCH%
-  - "conda install --file requirements.txt"
-  - "pip install twine codecov"
+  - "conda install -q --file requirements.txt"
+  - "pip install -q twine codecov"
   - "%CMD_IN_ENV% python setup.py build_ext -i --compiler=mingw32"
 
 # Not a .NET project, we build in the install step instead
@@ -69,7 +69,7 @@ build: false
 test_script:
   #- "%CMD_IN_ENV% ciocheck winpty --disable-tests"
   #- "conda remove pytest-xdist"
-  - "%CMD_IN_ENV% pytest winpty --cov=winpty --cov-report term-missing -v"
+  - "%CMD_IN_ENV% pytest winpty --cov=winpty -v"
   - "%CMD_IN_ENV% coveralls"
   - "%CMD_IN_ENV% codecov"
 

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -17,6 +17,10 @@ environment:
     APPVEYOR_RDP_PASSWORD: "dcca4c4863E30d56c2e0dda6327370b3#"
 
   matrix:
+    - PYTHON_VERSION: "3.7"
+      PYTHON_ARCH: "32"
+      ARCH: "x86"
+      platform: "x86"
     - PYTHON_VERSION: "3.6"
       PYTHON_ARCH: "32"
       ARCH: "x86"
@@ -29,6 +33,10 @@ environment:
       PYTHON_ARCH: "32"
       ARCH: "x86"
       platform: "x86"
+    - PYTHON_VERSION: "3.7"
+      PYTHON_ARCH: "64"
+      ARCH: "amd64"
+      platform: "x64"
     - PYTHON_VERSION: "3.6"
       PYTHON_ARCH: "64"
       ARCH: "amd64"

--- a/requirements.txt
+++ b/requirements.txt
@@ -8,6 +8,7 @@ m2w64-toolchain
 libpython
 
 # Test requirements
+pytest<4
 coveralls
 flaky
 pytest-cov

